File: app-kit/src/main/java/media/pixi/appkit/ui/chat/MessageListItem.java
package media.pixi.appkit.ui.chat;

import media.pixi.appkit.domain.chats.Message;

public class MessageListItem {

    public Message message;
    public MessageViewHolderType type;
    public boolean isMe;
    public String senderImageUrl;
    public float progress;

    public MessageListItem(Message message) {
        this.message = message;
    }

    public Message getMessage () {
        return message;
    }

    public MessageViewHolderType geMessageViewHolderType() {
        return type;
    }

    public long getTimeInMillis() {
        return message.getDate().toDate().getTime();
    }


}

File: app-kit/src/main/java/media/pixi/appkit/ui/chat/MessageListItem.kt
package media.pixi.appkit.ui.chat

import media.pixi.appkit.data.profile.UserProfile
import media.pixi.appkit.domain.chats.Message

data class MessageListItem(
    val id: String,
    val message: Message,
    val isMe: Boolean = false,
    val sendIconUrl: String = "",
    val senderId: String = "",
    val senderProfile: UserProfile?,
    val messageViewHolderType: MessageViewHolderType,
    val progress: Float = 0.toFloat(),
    val timeInMillis: Long = message.date.toDate().time,
    val hasSeen: Set<String> = emptySet()
)
